cmds/kexec/kexec_linux.go

```diff
@@ -135,21 +141,72 @@ func main() {
 		}
 	}
 
+	// mbk indicates that we are a multiboot kernel
+	var mbk bool
+	var kernelpath string
 	if opts.load {
-		kernelpath := flag.Args()[0]
+		kernelpath = flag.Args()[0]
 		log.Printf("Loading %s for kernel\n", kernelpath)
 
-		var l loader = file{opts.initramfs}
 		if err := multiboot.Probe(kernelpath); err == nil {
+			mbk = true
+		} else if err == multiboot.ErrFlagsNotSupported {
+			log.Fatal(err)
+		}
+	}
+
+	var segs []kexec.Segment // used to add other segments for, e.g., multiboot
+	if opts.acpi != "" {
+		if !mbk {
+			log.Fatalf("You can only specify -a with multiboot kernels")
+		}
+		// NewSDT won't work on some linux kernels that limit reading
```

Comment:
The inside of this if-statement could be moved to a function, maybe called `loadACPISegmentForMultiboot`.

Reply:
done